aptos-move/block-executor/src/modules_and_scripts.rs

```diff
@@ -121,8 +121,12 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> CodeStorage
         let immediate_dependencies = partially_verified_script
             .immediate_dependencies_iter()
             .map(|(addr, name)| {
-                self.fetch_verified_module(addr, name)
-                    .map_err(|e| expect_no_verification_errors(e))
+                if self.check_module_exists(addr, name)? {
+                    self.fetch_verified_module(addr, name)
```

Comment:
wouldn't fetch verified module also give an error? what's the rationale for checking existence with a separate call? (also it's not atomic right, unless captured reads like mechanism would make it consistent)

Reply:
Short answer, error status code.

 If we fetch, we try to verify the code based on the definitions inside the script, and if they do not exist, we return an "unverifiable code is stored" error. Basically, first load can always fail because of non-existence, but all other closure loads should not. My thinking was that the need to check for existence is nicer because you can then "assert" that `fetch_verified_module` should always succeed if the module is from storage.

Reply:
the delicate thing here is that unless we assume specific caching behavior, or that it's not based on speculative state, we can't really assert anything. However, I think I understand now that you want the proper error code behavior in the "correct / committed" execution - which this will provide as any reads, consistent with each other or not, are validated. However, we should be careful to not actually assert anything (I suppose that's why you wrote "assert"). If so, we should be good.